integrations/pinecone/src/haystack_integrations/document_stores/pinecone/document_store.py

```diff
@@ -267,6 +267,19 @@ def _embedding_retrieval(
 
         return self._convert_query_result_to_documents(result)
 
+    @staticmethod
+    def _convert_to_int(metadata: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Convert metadata original int values back to integers again, since PineCone store numbers as float.
```

Comment:
Let's explain we only do this for specific metadata fields used by some retrievers.
nit: "Pinecone" instead of "PineCone"